engineapi/engineapi/middleware.py

```diff
@@ -59,30 +143,33 @@ async def dispatch(
         if path in self.whitelist.keys() and self.whitelist[path] == method:
             return await call_next(request)
 
-        authorization_header = request.headers.get("authorization")
-        if authorization_header is None:
+        authorization = request.headers.get("authorization")
+        if authorization is None:
             return Response(
-                status_code=403, content="No authorization header passed with request"
+                status_code=403,
+                content="No authorization header passed with request",
             )
-        user_token_list = authorization_header.split()
-        if len(user_token_list) != 2:
+
+        try:
+            _, user_token = parse_auth_header(auth_header=authorization)
+        except InvalidAuthHeaderFormat:
             return Response(status_code=403, content="Wrong authorization header")
-        user_token: str = user_token_list[-1]
+        except Exception as e:
+            logger.error(f"Error processing Brood response: {str(e)}")
+            return Response(status_code=500, content="Internal server error")
 
         try:
-            user: BugoutUser = bc.get_user(user_token)
-            if not user.verified:
-                logger.info(
-                    f"Attempted journal access by unverified Brood account: {user.id}"
-                )
-                return Response(
-                    status_code=403,
-                    content="Only verified accounts can access journals",
-                )
-            if str(user.application_id) != str(MOONSTREAM_APPLICATION_ID):
-                return Response(
-                    status_code=403, content="User does not belong to this application"
-                )
+            user: BugoutUser = bugout_auth(token=user_token)
```

Comment:
I would expect to see an `if` statement like:

```python
if user_token is None:
    return call_next(request)
```

Reply:
Instead of letting the `bugout_auth` call fail.

Reply:
It could not be node, it could `""` and it will fall if it is empty at:
https://github.com/moonstream-to/api/blob/d3effd952cd0795883e5f5799dca2fff9074ac64/engineapi/engineapi/middleware.py#L62C21-L62C21

Reply:
So put that condition also in the if statement?

Reply:
It uses in `user: Optional[BugoutUser] = Depends(user_for_auth_header),` at line 302 in `metatx.py` where I handle user = `None` according with documentation [FastAPI Deps](https://fastapi.tiangolo.com/tutorial/dependencies/) it does not shows options where it `return call_next(request)` I think it is unsupported approach to use in FastAPI Dependencies. I can add
```python
if user_token != "":
  ...
else:
  return None
```
for clarification